odl/deform/optimal_information_transport.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Example of shape-based image reconstruction
+using optimal information transformation.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+from odl.operator.operator import Operator
+from builtins import super
+import numpy as np
+import matplotlib.pyplot as plt
+from odl.discr import Gradient
+from odl.trafos import FourierTransform
+from odl.util import snr
+standard_library.install_aliases()
+
+__all__ = ('optimal_information_transport_solver',)
+
+
+def optimal_information_transport_solver(gradS, I, niter, eps,
+                                         sigma, callback=None):
+
+    DPhiJacobian = gradS.domain.one()
+
+    grad = Gradient(gradS.domain, method='central')
+
+    # We solve poisson using the fourier transform
+    # ft = FourierTransform(op.domain)
+    ft = FourierTransform(gradS.domain)
+    k2_values = sum((ft.range.points() ** 2).T)
+    k2 = ft.range.element(np.maximum(np.abs(k2_values), 0.01))
+    poisson_solver = ft.inverse * (1 / k2) * ft
```

Comment:
This is currently quite a hack. What should be done is that we should set the fourier transform to 0 in the 0, that is, no DC component.

```python
k2_values = sum((ft.range.points() ** 2).T)
k2_values[k2_values == 0] = np.inf
k2 = ft.range.element(np.abs(k2_values))
poisson_solver = ft.inverse * (1 / k2) * ft
```